builtin/logical/jwt/path_roles.go

```diff
@@ -0,0 +1,197 @@
+package jwt
+
+import (
+	"fmt"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/fatih/structs"
+	jwt "github.com/dgrijalva/jwt-go"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathRoles(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: `roles/(?P<name>\w+)`,
+		Fields: map[string]*framework.FieldSchema{
+			"name": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Name of the Role",
+			},
+			"algorithm": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Default:     "RS256",
+				Description: "Algorithm for JWT Signing",
+			},
+			"key": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Private Key (RSA or EC) or String for HMAC Algorithm",
+			},
+			"lease": &framework.FieldSchema{
+				Type:        framework.TypeBool,
```

Comment:
I'd make this a time duration instead of bool. Since with global default lease, and soon per-backend default lease it may not always be 30 days. Also you may only want a 24h or 3h lease, etc, but something less than the max.